docs/source/extending.rst

```diff
@@ -0,0 +1,144 @@
+|                    |           |``"directory"``.              |
++--------------------+-----------+------------------------------+
+|**created**         |unicode    |Creation date of the entity,  |
+|                    |           |as ISO-8601 string.           |
+|                    |           |                              |
++--------------------+-----------+------------------------------+
+|**last_modified**   |unicode    |Last modified date of the     |
+|                    |           |entity, as ISO-8601 string.   |
+|                    |           |                              |
++--------------------+-----------+------------------------------+
+|**content**         |variable   |The "content" of the entity.  |
+|                    |           |(:ref:`See                    |
+|                    |           |Below<modelcontent>`)         |
++--------------------+-----------+------------------------------+
+|**mimetype**        |unicode or |The mimetype of ``"content"``,|
+|                    |``None``   |if any.  (:ref:`See           |
+|                    |           |Below<modelcontent>`)         |
++--------------------+-----------+------------------------------+
+|**format**          |unicode or |The format of ``"content"``,  |
+|                    |``None``   |if any. (:ref:`See            |
+|                    |           |Below<modelcontent>`)         |
++--------------------+-----------+------------------------------+
+
+.. _modelcontent:
+
+The **"content"** field
+
+In certain circumstances, we may not need the full content of an entity to
+complete a Contents API request.  require the full content of a filesystem
+entity.  For example, if we want to list the entries in directory ""foo/bar",
+we don't want to include the data stored in each file
+
+A sample model looks as follows:
+
+.. sourcecode:: python
+
+   {
+       "name": "My Notebook.ipynb",
+       "path": "foo/bar/My Notebook.ipynb",
+       "type": "notebook",
+       "writable": "true",
+       "created": "2013-10-01T14:22:36.123456+00:00",
+       "last_modified": "2013-10-02T11:29:27.616675+00:00",
+       "mimetype": None,
+       "content": None,
+       "format": None,
+   }
+
+Writing a Custom ContentsManager
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Required Methods
+''''''''''''''''
+
+A minimal complete implementation of a custom
+:class:`ContentsManager<manager.ContentsManager>` must implement the following
+methods:
+
+.. autosummary::
+   ContentsManager.get
+   ContentsManager.save
+   ContentsManager.delete_file
+   ContentsManager.rename_file
+   ContentsManager.file_exists
+   ContentsManager.dir_exists
+   ContentsManager.is_hidden
+
+Testing
+'''''''
+.. currentmodule:: notebook.services.contents.tests
+
+:mod:`notebook.services.contents.tests` includes several test suites written
+against the abstract Contents API.  This means that an excellent way to test a
+new ContentsManager subclass is to subclass our tests (**!**) to make them use
```

Comment:
This was mostly a note to myself while writing.  I do think that adding some amount of emphasis here is appropriate, since using tests as public API for extensions is a fairly novel concept as far as I know.